PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -149,14 +152,31 @@ public void windowOpened(WindowEvent e) {
     }
 
     private void updateAvailableAUAndTotalAU() {
-        ctrlProvider.getAvailableAUAndTotalAU().subscribe(pair -> {
-            availableAUField.setText(String.valueOf(pair.getLeft()));
-            totalAUField.setText(String.valueOf(pair.getRight()));
-        });
+        refreshButton.setEnabled(false);
+        ctrlProvider.getAvailableAUAndTotalAU()
+                .observeOn(ideaSchedulers.dispatchUIThread())
+                .doOnNext(pair -> {
+                    availableAUField.setText(String.valueOf(pair.getLeft()));
+                    totalAUField.setText(String.valueOf(pair.getRight()));
+                    refreshButton.setEnabled(true);
```

Comment:
You can just compare available and total AU here.

Reply:
We have to compare `availableAU` with `calculatedAU` rather than `totalAU` when we want to know if AU is enough

---
In reply to: [202249188](https://github.com/Microsoft/azure-tools-for-java/pull/1763#discussion_r202249188) [](ancestors = 202249188)

Reply:
Can these UI operations be put into `setData()`?